lib/types/array.js

```diff
@@ -226,15 +226,6 @@ module.exports = Any._extend({
                             --j;
                             --jl;
 
-                            if (!schema._flags.sparse &&
-                                res.value === undefined) {
-
-                                errors.push(error('array.sparse', { key, path, pos: i, value: undefined }, state.localize(path)));
-                                if (prefs.abortEarly) {
-                                    return errors;
-                                }
-                            }
```

Comment:
Wasn't sure about this one, but I cut this because Joi covers this branch only through [tests that use `extend`](https://github.com/sideway/joi/blob/master/test/types/array.js#L1767-L1830); since `extend`'s been cut (as far as I can tell?), i assumed this should go, too. My bad if I whiffed on that!

Reply:
I believe you can still trigger this condition using `custom()`, so we may want to bring it back and cover it.  Here's an example schema I tried out:
```js
            const schema = Joi.array().items(Joi.any().custom(() => undefined).required());
```

Reply:
Oh, gotcha. Totally overlooked, my bad! Thanks for that, pushing up a fix in just a few 🙏